docs/howto/how-to-make-api-tool/api-tool-guide.md

```diff
@@ -0,0 +1,341 @@
+[Generating config files] finished
+[Generating test files for APItool] started
+[Generating test files for APItool] .  Generate tests
+[Generating test files for APItool] finished
+```
+
+After this command finishes, you will see a file named `a_p_i_tool.py` under `~/backend/ayx_plugins/` with the boilerplate code. When you open the file, you should see something like this:
+
+```python
+    class APITool(PluginV2):
+        """Concrete implementation of an AyxPlugin."""
+
+        def __init__(self, provider: AMPProviderV2) -> None:
+            self.provider = provider
+            # truncated code
+
+        def on_incoming_connection_complete(self, anchor: namedtuple) -> None:
+            # truncated code
+
+        def on_record_batch(self, batch: "Table", anchor: namedtuple) -> None:
+            # truncated code
+
+        def on_complete(self) -> None:
+            import pandas as pd
+            import pyarrow as pa
+
+            df = pd.DataFrame(
+                {
+                    "x": [1, 2, 3],
+                    "y": ["hello", "world", "from ayx_python_sdk!"],
+                    "z": [self.config_value, self.config_value, self.config_value],
+                }
+            )
+
+            packet = pa.Table.from_pandas(df)
+
+            self.provider.write_to_anchor("Output", packet)
+            self.provider.io.info("APITool tool done.")
+```
+## Write API Request Tool
+---
+Now you are ready to modify this plugin code to pull data from an API and tell the plugin to output it! In our example, we use the [requests](https://requests.readthedocs.io/en/latest/) library to do this.
+
+> :information_source: Since this is an input tool, we only focus on the `on_complete` function. For additional information on the lifecycle of a plugin, refer to the [AYX Python SDK documentation](https://alteryx.github.io/ayx-python-sdk/plugin_lifecycle.html)
+
+### 1. Make the Request
+In this example, we fetch 2016 NBA playoff statistics data for Lebron James from the [BALLDONTLIE API](https://app.balldontlie.io)
+
+First, we want to remove the existing boilerplate code in the `on_complete` function, and leave only the `import pyarrow as pa` line. Your code should now look like this:
+
+```python
+    def on_complete(self) -> None:
+        import pyarrow as pa
+        import pyarrow.compute as pc # add this line in, it will be used later
+```
+
+Next, we write a function to make a GET request to the API. We pass in Lebron James' `player_id` and `season`:
+```python
+    def get_postseason_stats(player_id=237, season=2016):
+        resp = requests.get(
+            "https://www.balldontlie.io/api/v1/stats",
+            params={
+                "player_ids[]": player_id,
+                "postseason": True,
+                "seasons[]": season,
+            },
+        )
+        return resp.json()["data"]
+```
+When we call this function with `player_id=237` and `season=2016`, we get back the data as a JSON.
```

Comment:
@ayx-orion update "we get back the data as a JSON." to "we get the data back as a JSON."